account_customer_wallet/models/account_payment.py

```diff
@@ -0,0 +1,47 @@
+# Copyright (C) 2022-Today: GRAP (http://www.grap.coop)
+# @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, fields, models
+from odoo.exceptions import UserError
+
+
+class AccountPayment(models.Model):
+    _inherit = "account.payment"
+
+    is_customer_wallet_journal = fields.Boolean(
+        related="journal_id.is_customer_wallet_journal",
+    )
+
+    customer_wallet_balance = fields.Monetary(
+        string="Customer Wallet Balance",
+        currency_field="currency_id",
+        related="partner_id.customer_wallet_balance",
+    )
+
+    def action_post(self):
+        wallet_payments = self.filtered(
+            lambda x: x.journal_id.is_customer_wallet_journal
+        )
+
+        for payment in wallet_payments:
+            if (
+                payment.partner_id
+                and (payment.customer_wallet_balance - payment.amount)
+                < payment.journal_id.minimum_wallet_amount
+            ):
+                raise UserError(
```

Comment:
In manual testing, this check works, BUT it doesn't test whether any of the move lines actually credit the customer wallet account.